pkg/controller/certificates/authority/policies.go

```diff
@@ -44,14 +47,18 @@ type SigningPolicy interface {
 //  * It sets IsCA to false.
 //  * It validates that the signer has not expired.
 //  * It sets NotBefore and NotAfter:
+//    duration = min(TTL, ExpirationSeconds)
 //    All certificates set NotBefore = Now() - Backdate.
-//    Long-lived certificates set NotAfter = Now() + TTL - Backdate.
-//    Short-lived certificates set NotAfter = Now() + TTL.
+//    Long-lived certificates set NotAfter = Now() + duration - Backdate.
+//    Short-lived certificates set NotAfter = Now() + duration.
 //    All certificates truncate NotAfter to the expiration date of the signer.
 type PermissiveSigningPolicy struct {
 	// TTL is used in certificate NotAfter calculation as described above.
 	TTL time.Duration
 
+	// ExpirationSeconds is used in certificate NotAfter calculation as described above.
+	ExpirationSeconds *int32
```

Comment:
I'd suggest leaving PermissiveSigningPolicy alone and calculating the TTL of the cert before constructing PermissiveSigningPolicy. That saves you plumbing and teaching more layers about feature enablement